services/example-website/config/omeda-identity-x.js

```diff
@@ -106,4 +108,63 @@ module.exports = {
       promoCode: 'P1FullProfile',
     },
   ],
+
+  // onLoginLinkSentFormatter: (async ({ payload }) => ({ ...payload })),
+  onAuthenticationSuccessFormatter: (async ({ payload }) => ({ ...payload, promoCode: 'ExampleWebsiteOnAuthSuccessPromo' })),
+  onUserProfileUpdateFormatter: (async ({ req, payload }) => {
+    // BAIL if omedaGraphQLCLient isnt available return payload.
+    if (!req.$omedaGraphQLClient) return payload;
+
+    const idxOnProductHooks = req.app.locals.site.getAsObject('idxOnProductHooks');
+    const omeda = req.app.locals.site.getAsObject('omeda');
+    if (idxOnProductHooks.onUserProfileUpdate) {
+      const { productIds, promoCode } = idxOnProductHooks.onUserProfileUpdate;
+      const { user } = payload;
+      // Get the encriptedCustomerId that matches the omeda brandKey
+      const encryptedCustomerId = user.externalIds.filter(({
+        identifier,
+        namespace,
+      }) => identifier.type === 'encrypted'
+      && namespace.provider === 'omeda'
+      && namespace.tenant === omeda.brandKey)[0].identifier.value;
+
+      // BAIL if no encryptedCustomerId and return payload
+      if (!encryptedCustomerId) return payload;
```

Comment:
Use an object-path helper here, if the user doesn't have any matching external ids, this will error. Also if you only want the first matching value, use `find` instead of `filter` because it will stop iterating once it finds a match.
```suggestion
      const found = getAsArray(user, 'externalIds')
        .find(({ identifier, namespace }) => identifier.type === 'encrypted'
          && namespace.provider === 'omeda'
          && namespace.tenant === omeda.brandKey
        );
      // BAIL if no encryptedCustomerId and return payload
      if (!found) return payload;
      const encryptedCustomerId = get(found, 'identifier.value');
```